Src/VimCore/InsertUtil.fs

```diff
@@ -208,70 +208,46 @@ type internal InsertUtil
                     let span = column.Column.Span
                     textEdit.Replace(span.Span, text) |> ignore
 
-            if textEdit.HasEffectiveChanges then 
+            if textEdit.HasEffectiveChanges then
                 textEdit.Apply() |> ignore
             else
                 textEdit.Cancel()
 
-        // Actually apply the edit to the lines in the block selection
-        let doApply () = 
-            use textEdit = _textBuffer.CreateEdit()
-            let mutable abortChange = false
-
-            let currentSnapshot = x.CurrentSnapshot
-            for i = 0 to (height - 1) do
+        // Actually apply the edit to the lines in the block selection.
+        let doApply () =
+            for i = 0 to height - 1 do
 
+                let currentSnapshot = x.CurrentSnapshot
                 let lineNumber = startLineNumber + i
                 let currentLine = SnapshotUtil.GetLine currentSnapshot lineNumber
 
-                // Only apply the edit to lines which were included in the original selection
+                // Only apply the edit to lines which were included in the
+                // original selection.
                 let tabStop = _localSettings.TabStop
                 let column =
                     if atEndOfLine then SnapshotColumn.GetLineEnd(currentLine) |> Some
                     else SnapshotColumn.GetColumnForSpaces(currentLine, spaces, tabStop)
-                match column with 
+                match column with
                 | Some column ->
-                    let position = column.StartPosition
-                    let text =
-                        if _localSettings.ExpandTab then
-                            StringUtil.ExpandTabsForColumn text column.ColumnNumber tabStop
-                        else
-                            text
-
-                    // Unfortunately, we may have to delete some of the preceding
-                    // spaces if they are no longer needed to maintain the
-                    // column position.
-                    if text.StartsWith("\t") then

```

Comment:
Did you verify that we have block insert tests that cover this scenario?

Reply:
I did not check, but the code itself is just trying to mimic what would automatically occur if a tab insert command were run. In other words, the old code was inserting a literal string, but realized that if they user had actually typed tab as part of an insert, it would (as vim does) "fix up" any no longer necessary previous spaces.

This is another benefit of repeating the insert instead of inserting strings: this behavior occurs naturally as part of the insertion command instead of being hacked into block insertion with special (and duplicated) handling.

I just confirmed that it does indeed work correctly. I will check whether there is coverage for this scenario in any of the existing tests.

Reply:
OK, I checked and there is a test:

```csharp
                /// <summary>
                /// A block inserted tab with 'noexpandtab' should eat preceding spaces if possible
                /// </summary>
                [WpfFact]
                public void UnalignedNoExpandTabs()
```